social/src/main/java/bisq/social/userprofile/BsqTxValidator.java

```diff
@@ -0,0 +1,100 @@
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.social.userprofile;
+
+import bisq.common.data.Pair;
+import bisq.common.encoding.Hex;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BsqTxValidator {
+
+    public static boolean initialSanityChecks(String txId, String jsonTxt) {
+        if (jsonTxt == null || jsonTxt.length() == 0) {
+            return false;
+        }
+        JsonObject json = new Gson().fromJson(jsonTxt, JsonObject.class);
+        // there should always be "id" string element at the top level
+        if (json.get("id") == null) {
+            return false;
+        }
+        // txid should match what we requested
+        if (!txId.equals(json.get("id").getAsString())) {
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean isBsqTx(String url, String txId, String jsonTxt) {
+        return url.matches(".*bisq.*") && initialSanityChecks(txId, jsonTxt);
+    }
+
+    public static boolean isProofOfBurn(String jsonTxt) {
+        JsonObject json = new Gson().fromJson(jsonTxt, JsonObject.class);
+        JsonElement txType = json.get("txType");
+        if (txType == null) {
+            return false;
+        }
+        return txType.getAsString().equalsIgnoreCase("PROOF_OF_BURN");
+    }
+
+    public static boolean isLockup(String jsonTxt) {
+        JsonObject json = new Gson().fromJson(jsonTxt, JsonObject.class);
+        JsonElement txType = json.get("txType");
+        if (txType == null) {
+            return false;
+        }
+        return txType.getAsString().equalsIgnoreCase("LOCKUP");
+    }
+
+    public static byte[] getOpReturnData(String jsonTxt) {
+        try {
+            Pair<JsonArray, JsonArray> vinAndVout = getVinAndVout(jsonTxt);
+            JsonArray voutArray = vinAndVout.second();
+            for (JsonElement x : voutArray) {
+                JsonObject y = x.getAsJsonObject();
+                if (y.get("txOutputType").getAsString().matches(".*OP_RETURN.*")) {
+                    String opReturnHexString = y.get("opReturn").getAsString();
+                    return Hex.decode(opReturnHexString);
+                }
+            }
+        } catch (JsonSyntaxException e) {
+            log.error("json error:", e);
+        }
+        return null;
```

Comment:
We try to avoid null as far as possible in Bisq 2. Better propagate the exception if the exception is not an expected case or return Optional.